From f8b87376ce9c48f4ec1348610fddcb479650e2d9 Mon Sep 17 00:00:00 2001
From: afl1 <afl2001@gmail.com>
Date: Wed, 7 Aug 2019 04:02:23 +0100
Subject: [PATCH 15/43] VideoPlayer: dirty fix for audio stall detection

Patch: kodi-le-042-dirty-fix-for-audio-stall-detection.patch
---
 xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp b/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp
index 68d26ee..972cb26 100644
--- a/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp
+++ b/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp
@@ -275,9 +275,9 @@ void CVideoPlayerAudio::Process()
           !m_stalled && m_syncState == IDVDStreamPlayer::SYNC_INSYNC)
       {
         // while AE sync is active, we still have time to fill buffers
-        if (m_syncTimer.IsTimePast())
+        if (m_syncTimer.IsTimePast() && m_audioClock < m_pClock->GetClock())
         {
-          CLog::Log(LOGNOTICE, "CVideoPlayerAudio::Process - stream stalled");
+          CLog::Log(LOGNOTICE, "CVideoPlayerAudio::Process - stream stalled pts:%0.3f clock:%0.3f ", m_audioClock / DVD_TIME_BASE, m_pClock->GetClock() / DVD_TIME_BASE);
           m_stalled = true;
         }
       }
-- 
2.7.4

